### DISCUSSION TITLE
Satellite imagery object detection

### DISCUSSION THREAD

**Discussion by neptyko** [ultralytics/ultralytics#17745] [Q&A]
Hi,
I plan to use the YOLOv11 model in my research, which will focus on detecting various objects in commercial satellite imagery.

One of the main questions I have is about the enormous size of these images—approximately 50,000 by 50,000 pixels. How should I prepare the dataset to ensure YOLOv11 can work with it effectively? Should I annotate objects directly on such large images, or would it be better to split them into smaller tiles, e.g., 10,000 by 10,000 pixels? Does YOLOv11 have any limitations regarding image dimensions, both during training and detection?

Additionally, I plan to work with panchromatic images (training and testing the model). Could this potentially reduce the model's performance?

I would appreciate your help. Best regards.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @neptyko, thank you for reaching out and for your interest in using YOLOv11 for satellite imagery detection 🚀! This is an automated response, but rest assured, an Ultralytics engineer will assist you soon.

In the meantime, we encourage you to explore our [Docs](https://docs.ultralytics.com) for guidance on handling large image datasets and the model's capabilities. If this is a 🐛 Bug Report, please include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help expedite debugging.

For questions regarding custom training, it would be helpful if you could provide additional information like dataset samples or training logs. This detail can assist in giving you a more precise answer. Additionally, ensure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Community Support

Feel free to join us on [Discord](https://ultralytics.com/discord) for real-time discussions 🎧, check out [Discourse](https://community.ultralytics.com) for deeper conversations, or share insights on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Remember to keep your `ultralytics` package updated to the latest version along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in the following verified environments, ensuring all dependencies are pre-installed:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Please refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Refer to the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If the badge is green, this indicates that all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. The CI confirms the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across different operating systems.

Your understanding and patience are appreciated as we work on your query! 😊

**Comment by glenn-jocher** [comment]
@neptyko to effectively use YOLO11 with large satellite images, it's recommended to divide the images into smaller, manageable tiles (e.g., 10,000x10,000 pixels) and annotate objects within these tiles, as YOLO11 may not handle extremely large dimensions like 50,000x50,000 pixels efficiently due to memory constraints and training limitations. For panchromatic images, ensure proper normalization and consider fine-tuning a pretrained model to address differences in color space, as limited image channels can potentially impact performance. You can find more guidance on dataset preparation and training in the [YOLO11 training guide](https://docs.ultralytics.com/modes/train/).

**Comment by neptyko** [comment]
OK, so I have a couple of questions that I couldn't find answers to in the tutorial.
1. what is the maximum size in pixels that YOLOv11 can support?
2. what is the maximum weight of images in mb or GB that YOLOv11 can handle? ( Weight of training dataset)
3. how to optimise panchromatic imaging? 
Thank you in advance, regards :)

**Reply by glenn-jocher** [reply]
@neptyko thank you for your questions. 

1. YOLOv11 does not strictly limit image size but working with excessively large images (e.g., 50,000x50,000) can lead to high memory consumption; splitting images into smaller tiles for training is recommended.  
2. The maximum dataset size or image weight is constrained by your hardware's memory (RAM/GPU) capacity. Ensure your system can load batches during training without memory overflow.  
3. For panchromatic imaging, consider preprocessing with contrast enhancement or histogram equalization to optimize feature extraction before training.  

If you need further details, feel free to consult the YOLO11 documentation: https://docs.ultralytics.com/models/yolo11/.

**Reply by neptyko** [reply]
OK, and what do you think of this approach to test data preparation?
It labels objects on the ,,large,,images and then uses code that will cut these objects in the image according to the position of the bounding boxes. I will get the images together with the labels of about 200x160 pixels. I will upload these to YOLOv11 training, and then with the model trained on this I will test on 50k x 50k pixel images. Is this a good approach? Does YOLOv11 require this to be tested on the same image dimensions from which I trained it?

**Reply by glenn-jocher** [reply]
@neptyko your approach of labeling objects on large images, cropping them based on bounding boxes, and training YOLOv11 on smaller tiles (e.g., 200x160 pixels) is valid and commonly used for high-resolution datasets. YOLOv11 does not require testing on the same image dimensions as the training set. During testing, as long as the objects in the large images are within the resolution range where the model can generalize well, YOLOv11 will handle inference effectively.
